e2etest/bgptests/bgp.go

```diff
@@ -1209,6 +1209,185 @@ var _ = ginkgo.Describe("BGP", func() {
 				false,
 				ipfamily.IPv6))
 
+		table.DescribeTable("A service IP will not be advertised to peers outside the BGPAdvertisement peers list", func(addressRange string, ipFamily ipfamily.Family) {
+			var testContainers []*frrcontainer.FRR
+			for _, c := range FRRContainers {
+				if !strings.Contains(c.Name, "multi") {
+					testContainers = append(testContainers, c)
+				}
+			}
+
+			if len(testContainers) < 2 {
+				ginkgo.Skip("This test requires 2 external frr containers")
+			}
+
+			resources := metallbconfig.ClusterResources{
+				Peers: metallb.PeersForContainers(FRRContainers, ipFamily),
+			}
+
+			for _, c := range FRRContainers {
+				err := frrcontainer.PairWithNodes(cs, c, ipFamily)
+				framework.ExpectNoError(err)
+			}
+
+			err := ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			for _, c := range FRRContainers {
+				validateFRRPeeredWithNodes(cs, c, ipFamily)
+			}
+
+			allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			framework.ExpectNoError(err)
+
+			var pools []metallbv1beta1.IPAddressPool
+			var services []*corev1.Service
+			for i := 0; i < 2; i++ {
```

Comment:
I would unroll the loop and create resources upfront.
Two pools, two advertisements on one pool each, the two advertisements paired to two different peers upfront, and then we validate